app/views/users/index.html.erb

```diff
@@ -1,2 +1,10 @@
-<h1>users</h1>
-
+<h1>Users</h1>
+<div class="user-card">
+  <% @users.each do |user| %>
+    <%= link_to user_path(user.id) do %>
+      <div class="user-repeat">
+        <%= render partial: "shared/username", locals: { user: user } %>
+      </div>
+    <% end %>
+  <% end %>
```

Comment:
[OPTIONAL]

- It is not a good practice to write logic or loop in views. I have shown you below what is the better practice instead of using `@objects.each`.

```ruby
    # app/views/users/_user.html.erb

     #   HTML Code Here displaying
     #   user.photo, user.name and user.posts_counter

    # app/views/users/index.html.erb

    <%= render @users %>
```

Reply:
I am wondering how this code will work! did you try it?

Reply:
It will work, but for that, you have to make a partial in view as what I mentioned above.

Reply:
I already created a partial for this purpose in app/views/shared/_username.html.erb

Reply:
As I have mentioned above, You should create a partial here. `app/views/users/_user.html.erb`